src/core/router/info.cc

```diff
@@ -715,23 +694,20 @@ void RouterInfo::CreateBuffer(const PrivateKeys& private_keys)
       // Create RI
       core::StringStream router_info;
       CreateRouterInfo(router_info, private_keys);
-      if (router_info.Str().size() > Size::MaxBuffer)
-        throw std::length_error("created RI is too big");
 
       // Create buffer
-      m_BufferLen = router_info.Str().size();
-      if (!m_Buffer)
-        m_Buffer = std::make_unique<std::uint8_t[]>(Size::MaxBuffer);
-      std::memcpy(m_Buffer.get(), router_info.Str().c_str(), m_BufferLen);
+      m_Buffer(
+          reinterpret_cast<const std::uint8_t*>(router_info.Str().c_str()),
+          router_info.Str().size());
 
       // Signature
       // TODO(anonimal): signing should be done when creating RI, not after. Requires other refactoring.
       private_keys.Sign(
-          reinterpret_cast<std::uint8_t*>(m_Buffer.get()),
-          m_BufferLen,
-          reinterpret_cast<std::uint8_t*>(m_Buffer.get()) + m_BufferLen);
+          m_Buffer.data(),
+          m_Buffer.size(),
+          m_Buffer.data() + m_Buffer.size());
```

Comment:
These changes are definitely better that what's currently in place.

This is potentially dangerous if `router_info.Str().size() + private_keys.GetPublic().GetSignatureLen()` is greater than `Size::MaxBuffer`, since it would write past the end of the buffer, without throwing an error. With the changes in #917, this will be a non-issue here, but may come up in other places.

See my comment about a suggested test for proof of a `SIGABRT`.

Reply:
This bug pre-existed the buffer util changes, and buffer util makes the situation better. With the changes in the signature PR, on top of this PR, this bug will no longer exist.

Reply:
Yes, this is a dangerous and stupid way to do things, but this applies to any underlying pointer handling (and this is legacy code btw). I imagined you'd rebase #917 and that's good because RI needs further refactoring.

Reply:
> I imagined you'd rebase #917 and that's good because RI needs further refactoring.

This is exactly what I intend to do. Happy to be helping fix the legacy code.